### PR TITLE
Convert static initialization into a single loop

### PR DESCRIPTION
**Pull request by DanHeidinga**
The code needs to loop over the static fields and initialize any
of them that have a constant value (J9FieldFlagConstant).  Original
code use three loops - one to process objects, one to process single
slot statics, one to process double slot statics - which meant
iterating over the romClass fields 3 times.

The new algorithm uses the counts in the J9ROMClass to figure out
the starting area for each of the 3 kinds of static fields.  This
saves two iterations over the fields (an expensive operation!)
while still maintaining the same layout of object/singles/doubles.

Keep track of whether the class being processed has static
fields with constantValueAttributes.  These fields require
extra work when initializing the class to ensure that the
constant values are copied into the J9Class->ramStatics.

Having this info allows optimizing class initialization to
skip this processing if the class doesn't have the field

### CODE REVIEW

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test extended xlinux jdk12

File: runtime/bcutil/ClassFileOracle.hpp [hunk 1]
```diff
@@ -924,6 +924,10 @@ class NameAndTypeIterator
 	bool annotationRefersDoubleSlotEntry() const { return _annotationRefersDoubleSlotEntry; }
 	bool isInnerClass() const { return _isInnerClass; }
 
+	bool needsStaticConstantInit() const {
```
Comment (gacholio) [inline runtime/bcutil/ClassFileOracle.hpp]: Should probably be formatted like the rest of the getters.

Comment (DanHeidinga) [reply runtime/bcutil/ClassFileOracle.hpp]: I've pushed a new commit with adjusted formatting

**Comment by gacholio** [comment]
CI build machines are misconfigured - failure isn't real.

**Pull request merged**